test/LoadURLTest.cpp

```diff
@@ -0,0 +1,246 @@
+// `getResultFunction` needed by the `Service` operation.
+class LoadURLTest : public ::testing::Test {
+ protected:
+  // Query execution context (with small test index) and allocator for testing,
+  // see `IndexTestHelpers.h`. Note that `getQec` returns a pointer to a static
+  // `QueryExecutionContext`, so no need to ever delete `testQec`.
+  QueryExecutionContext* testQec = ad_utility::testing::getQec();
+  ad_utility::AllocatorWithLimit<Id> testAllocator =
+      ad_utility::testing::makeAllocator();
+
+  // Factory for generating mocks of the `sendHttpOrHttpsRequest` function that
+  // is used by default by a `Service` operation (see the constructor in
+  // `Service.h`). Each mock does the following:
+  //
+  // 1. It tests that the request method is POST, the content-type header is
+  //    `application/sparql-query`, and the accept header is
+  //    `text/tab-separated-values` (our `Service` always does this).
+  //
+  // 2. It tests that the host and port are as expected.
+  //
+  // 3. It tests that the post data is as expected.
+  //
+  // 4. It returns the specified JSON.
+  //
+  // NOTE: In a previous version of this test, we set up an actual test server.
+  // The code can be found in the history of this PR.
+  static auto constexpr getResultFunctionFactory =
+      [](std::string predefinedResult,
+         boost::beast::http::status status = boost::beast::http::status::ok,
+         std::string contentType = "text/turtle",
+         std::exception_ptr mockException = nullptr,
+         ad_utility::source_location loc =
+             ad_utility::source_location::current())
+      -> LoadURL::GetResultFunction {
+    return [=](const ad_utility::httpUtils::Url&,
+               ad_utility::SharedCancellationHandle,
+               const boost::beast::http::verb& method,
+               std::string_view postData, std::string_view contentTypeHeader,
+               std::string_view acceptHeader) {
+      auto g = generateLocationTrace(loc);
+      EXPECT_THAT(method, testing::Eq(boost::beast::http::verb::get));
+      EXPECT_THAT(contentTypeHeader, testing::Eq(""));
+      EXPECT_THAT(acceptHeader, testing::Eq(""));
+      EXPECT_THAT(postData, testing::Eq(""));
+
+      if (mockException) {
+        std::rethrow_exception(mockException);
+      }
+
+      auto randomlySlice =
+          [](std::string result) -> cppcoro::generator<ql::span<std::byte>> {
+        // Randomly slice the string to make tests more robust.
+        std::mt19937 rng{std::random_device{}()};
+
+        const std::string resultStr = result;
+        std::uniform_int_distribution<size_t> distribution{
+            0, resultStr.length() / 2};
+
+        for (size_t start = 0; start < resultStr.length();) {
+          size_t size = distribution(rng);
+          std::string resultCopy{resultStr.substr(start, size)};
+          co_yield ql::as_writable_bytes(ql::span{resultCopy});
```

Comment:
The comment suggests that this is at least partially duplicated from the tests for SERVICE. If so, please deduplicate.